custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go

```diff
@@ -334,9 +370,21 @@ func (p *StackdriverProvider) GetExternalMetric(ctx context.Context, namespace s
 	if err != nil {
 		return nil, err
 	}
-	return &external_metrics.ExternalMetricValueList{
+
+	resp := &external_metrics.ExternalMetricValueList{
 		Items: externalMetricItems,
-	}, nil
+	}
+
+	if p.externalMetricCache != nil {
+		key := cacheKey{
+			namespace:      namespace,
+			metricSelector: metricSelector.String(),
+			info:           info,
+		}
+		p.externalMetricCache.Add(key, resp, p.externalMetricsCacheTTL)
```

Comment:
let's also make sure to use a mutex here as I don't believe this cache is thread safe

it is possible for HPA to be ran in parallel which could cause us to write to this from multiple threads at the same time

Reply:
I think its thread-safe actually https://github.com/kubernetes/apimachinery/blob/v0.32.3/pkg/util/cache/lruexpirecache.go#L106-L107 ?

Reply:
ah perfect, thanks!